### DISCUSSION TITLE
Stuck inside a wall after exiting ruins

### DISCUSSION THREAD

**Discussion by Kreymox** [FakeFishGames/Barotrauma#15141] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

one exit from ruins was leading right into a rock, i was teleported into it and tuck into textures (i tried to help myself with another char, but he stuck exactly the same). i made it out with a glider on 1 char, but another one remained there forever :( upd. all enemies close to that location are being magnetized to this place or smth - they don't attack me and just swim into the closest wall to this anomaly

### Reproduction steps

1. start any ruins scanning mission 2. go into them and try to find an exit leading right into the rock textures 3. be forced into the rock and die (you wont be able to move)

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Can you zip that save and put here? Any guess how often do ruins have exits to a wall?

> but another one remained there forever :( 

You should have used cheats, you could reenable achivements later if you care

> this anomaly

That's not an anomaly, ruins and walls exist on different planes and when you exit ruins you get teleported into level wall -> get stuck

Also i'm not a dev, but seems hard to fix, like, player could drill his way out (in) a wall anywhere, and if you force ruins to not intersect walls there might be no place for them, only in abyss

Perhaps i should just create another unstuck command, but for main character which won't be a cheat
![barodev small](https://github.com/user-attachments/assets/990f7597-ab76-4621-8710-3873bd425d39)

**Reply by Kreymox** [reply]
sure, here you go: https://drive.google.com/drive/folders/19epP08clYKmC-NylI2lOUhV0u9Ob8VEi?usp=drive_link
the route was Autonoe Linea - Harmonia Linea, you can load pre-last save
ruins are like 700-800m away from the starting point
as soon as you come you notice 2 doors - up and left - bug takes it's place in the upper one
-------
here is a video that shows the problem itself: https://drive.google.com/file/d/1UZfCD6-rWkq25WFaUuWRWCLXPGe4wIaT/view?usp=drive_link
![Screenshot](https://github.com/user-attachments/assets/ab1d09a1-1698-49df-a4d7-08455c42c5d4)

**Comment by SomeRandomNoobKekeke** [comment]
The ruins:
![20241114021926_1](https://github.com/user-attachments/assets/a3b708e2-8533-43de-bcd4-5ffcb1d65980)
Also that's not "natural" ruins, they are from scan ruins mission, there are no ruins on reverse path
But i don't think that matters

**Reply by Kreymox** [reply]
Ty, so we have just to accept this bug until it gets fixed? (if it gets)

**Reply by SomeRandomNoobKekeke** [reply]
Yes, and it looks like there is a logic that removes all wall vertices behind ruins rect, but devs forgot this 'edge' case
![BaroDev (wide)](https://github.com/user-attachments/assets/ef272e47-8689-4cbb-a404-00cf61b61924)

**Reply by Regalis11** [reply]
@Kreymox We will look into the issue, I unfortunately can't make any promises on it being fixed in the next update, but it _seems_ like a relatively simple thing to fix, especially now that we have a save file in which the issue occurs consistently. :)

@SomeRandomNoobKekeke I'm not sure what you mean by this being an edge case. There should be logic that removes the level walls that intersect with the ruin, and since that's exactly what's happening here, there must be something wrong in that removal logic.

**Reply by SomeRandomNoobKekeke** [reply]
@Regalis11 I mean, i didn't see the code, but if i would code it i would just remove vertices behind ruins rect and in this case there are no vertices, there just one giant edge - thus edge case

**Reply by Regalis11** [reply]
That's not quite how it works: if any of the edges of the wall intersect with the ruins, it should remove the whole cell (but that's not happening here, so something must be bugged). Relevant code: https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaShared/SharedSource/Map/Levels/Level.cs#L2296-L2309

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly, fix coming in the next update. Closing discussion.
